base/error.jl

```diff
@@ -60,12 +60,38 @@ Get a backtrace object for the current program point.
 """
 backtrace() = ccall(:jl_backtrace_from_here, Array{Ptr{Void},1}, (Int32,), false)
 
+struct InterpreterIP
+    code::CodeInfo
+    stmt::Csize_t
+end
+
 """
     catch_backtrace()
 
 Get the backtrace of the current exception, for use within `catch` blocks.
 """
-catch_backtrace() = ccall(:jl_get_backtrace, Array{Ptr{Void},1}, ())
+function catch_backtrace()
```

Comment:
We should be able to just avoid exposing the bare pointer information?

Reply:
Yeah, we can probably return two arrays here and keep a sentinel value to decide where to insert them.